### PR TITLE
M3-1506 Add Pagey to Images Landing

### PR DESCRIPTION
**Pull request by martinmckenna**
### Purpose

Add Pagey to ImagesLanding

### Notes

* Test w/ _prod-test-001_
* Changed the condition inside Pagey's `onDelete`
* Added some documentation about `onDelete` and why we're displaying toasts for image events

### CODE REVIEW

**Comment by martinmckenna** [comment]
^ working on that now

File: src/components/Pagey/Pagey.ts [hunk 1]
```diff
@@ -56,8 +57,18 @@ export default (requestFn: PaginatedRequest) => (Component: React.ComponentType<
     }
 
     private onDelete = () => {
-      const { data, page } = this.state;
-      if (data && data.length === 1) {
+      const { page, pages } = this.state;
+      /*
+       * Basically, if we're on page 3, but the API data
+       * only has 2 pages of data, we need to request
+       * the "pages" data from the API, which is the max number of pages.
```
Comment (martinmckenna) [inline src/components/Pagey/Pagey.ts]: how many times can I use "data" in a sentence

File: src/features/Images/ImagesLanding.tsx [hunk 2]
```diff
@@ -21,7 +21,8 @@ import { DocumentTitleSegment } from 'src/components/DocumentTitle';
 import ErrorState from 'src/components/ErrorState';
 import Grid from 'src/components/Grid';
 import Notice from 'src/components/Notice';
-import PaginationFooter, { PaginationProps } from 'src/components/PaginationFooter';
+import paginate, { PaginationProps } from 'src/components/Pagey';
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: It's called Pagey, Marty... (jk, dont care)

File: src/features/Images/ImagesLanding.tsx [hunk 3]
```diff
@@ -213,7 +205,8 @@ class ImagesLanding extends React.Component<CombinedProps, State> {
     deleteImage(this.state.removeDialog.imageID)
       .then(() => {
         this.closeRemoveDialog();
-        this.requestImages();
+        /* request generated by the Pagey HOC */
+        this.props.request();
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: Should we be using `this.props.onDelete()`?

File: src/features/Images/ImagesLanding.tsx [hunk 4]
```diff
@@ -343,8 +343,8 @@ class ImagesLanding extends React.Component<CombinedProps, State> {
         </Paper>
         <PaginationFooter
           count={count}
-          handlePageChange={this.handlePageChanage}
-          handleSizeChange={this.handlePageSizeChanage}
+          handlePageChange={this.props.handlePageChange}
```
Comment (rjstires) [inline src/features/Images/ImagesLanding.tsx]: Can you add handlePageChange and handlePageSizeChange to the destructured this.props at the top?

File: src/components/Pagey/Pagey.ts [hunk 5]
```diff
@@ -56,8 +57,18 @@ export default (requestFn: PaginatedRequest) => (Component: React.ComponentType<
     }
 
     private onDelete = () => {
```
Comment (johnwcallahan) [inline src/components/Pagey/Pagey.ts]: @rjstires added an `onDelete` handler, which has been merged to develop: https://github.com/linode/manager/pull/3946/commits/b2709f672f3fb558941650419a170c0161733cef

File: src/components/Pagey/Pagey.ts [hunk 6]
```diff
@@ -56,8 +57,18 @@ export default (requestFn: PaginatedRequest) => (Component: React.ComponentType<
...
+       * only has 2 pages of data, we need to request
+       * the "pages" data from the API, which is the max number of pages. 
+       * To the user, we're just automatically sending them to the previous page.
+       * 
+       * This solves the issue where the user deletes the last entity
+       * on a page and then sees an empty state instead of going to the
+       * previous page
+       */
+      if (pages && page > pages) {
         return this.handlePageChange(page - 1);
```
Comment (Jskobos) [inline src/components/Pagey/Pagey.ts]: Based on the comment above, I'd expect this to be `this.handelPageChange(pages)`. Hard to imagine a situation where the code you have doesn't work, though.

File: src/components/Pagey/Pagey.ts (hunk 5 above)
Comment (martinmckenna) [reply src/components/Pagey/Pagey.ts]: yep. We're using this in the deletion event. Check `componentDidMount`. I also added a comment to explain why I'm running `this.props.request` if a user triggers a deletion

File: src/components/Pagey/Pagey.ts (hunk 6 above)
Comment (martinmckenna) [reply src/components/Pagey/Pagey.ts]: 🤷‍♂️

**Pull request merged**